Gems/Terrain/Code/Source/TerrainRenderer/Components/TerrainSurfaceMaterialsListComponent.cpp

```diff
@@ -60,25 +59,49 @@ namespace Terrain
         if (serialize)
         {
             serialize->Class<TerrainSurfaceMaterialsListConfig, AZ::ComponentConfig>()
-                ->Version(1)
+                ->Version(2)
+                ->Field("DefaultMaterial", &TerrainSurfaceMaterialsListConfig::m_defaultSurfaceMaterial)
                 ->Field("Mappings", &TerrainSurfaceMaterialsListConfig::m_surfaceMaterials);
 
             AZ::EditContext* edit = serialize->GetEditContext();
             if (edit)
             {
                 edit->Class<TerrainSurfaceMaterialsListConfig>(
                         "Terrain Surface Material List Component", "Provide mapping between surfaces and render materials.")
+                    ->SetDynamicEditDataProvider(&TerrainSurfaceMaterialsListConfig::GetDynamicData)
                     ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
-                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::Show)
-                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                        ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::Show)
+                        ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
 
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &TerrainSurfaceMaterialsListConfig::m_defaultSurfaceMaterial,
+                        "Default Material", "The default material to fall back to where no other material surface mappings exist.")
                     ->DataElement(
                         AZ::Edit::UIHandlers::Default, &TerrainSurfaceMaterialsListConfig::m_surfaceMaterials,
                         "Material Mappings", "Maps surfaces to materials.");
             }
         }
     }
 
+    TerrainSurfaceMaterialsListConfig::TerrainSurfaceMaterialsListConfig()
+    {
+       m_hideSurfaceTagData.m_attributes.push_back(
+           {
+               AZ::Edit::Attributes::Visibility,
+               aznew AZ::Edit::AttributeData<AZ::Crc32>(AZ::Edit::PropertyVisibility::Hide)
+           }
+       );
+    }
+
+    const AZ::Edit::ElementData* TerrainSurfaceMaterialsListConfig::GetDynamicData(const void* handlerPtr, const void* elementPtr, const AZ::Uuid& )
+    {
+        const TerrainSurfaceMaterialsListConfig* owner = reinterpret_cast<const TerrainSurfaceMaterialsListConfig*>(handlerPtr);
+        if (elementPtr == &owner->m_defaultSurfaceMaterial.m_surfaceTag)
+        {
+            return &owner->m_hideSurfaceTagData;
+        }
+        return nullptr;
+    }
```

Comment:
Agreed, long term I want all the terrain components to separate out their editor and runtime functionality. I'd like to use the component structure we developed for Atom. In the short term though I decided to draw the line here since I think putting SetDynamicEditDataProvider() on the edit context of the editor component would force me to pull out all the edit context code from EditorWrappedComponentBase and do it directly in EditorTerrainSurfaceMaterialsListComponent... at that point might as well just switch to Atom's component model.